jcl/src/java.base/share/classes/openj9/internal/foreign/abi/InternalDowncallHandler.java

```diff
@@ -414,10 +461,14 @@ public InternalDowncallHandler(MethodType functionMethodType, FunctionDescriptor
 		scopeHandleMap = new ConcurrentHashMap<>();
 		/*[ENDIF] JAVA_SPEC_VERSION == 17 */
 
+		/*[IF JAVA_SPEC_VERSION >= 22]*/
+		threadHeapArgInfoMap = new ConcurrentHashMap<>();
```

Comment:
```java
		heapArgInfo = new ThreadLocal<>();
```

Reply:
Will update all the code around related to `ThreadLocal`.

Reply:
Updated the code with `ThreadLocal` and verified in tests.